OneSchedule.Domain/StateMachine/StateContext.cs

```diff
@@ -0,0 +1,81 @@
+﻿using OneSchedule.Data.Abstractions;
+using OneSchedule.Domain.Abstractions.StateMachine;
+using OneSchedule.Domain.Models;
+using OneSchedule.Entities;
+using OneSchedule.Helpers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OneSchedule.Domain.StateMachine
+{
+    public class StateContext : IStateContext
+    {
+        private const string NextState = "EventMenu";
+        private IState _state;
+        private readonly IRepository<ContextEntity> _contextRepository;
+        private readonly Dictionary<string, IState> _states;
+        public EventEntity EventEntity { get; set; }
+        public bool IsContextCompleted => EventEntity.StartDate != default(DateTime)
+                                          && string.IsNullOrWhiteSpace(EventEntity.Title)
+                                          && EventEntity.Notifications is
+                                          {
+                                              Count: > 0
+                                          };
+
+        public StateContext(IEnumerable<IState> states, IRepository<ContextEntity> contextRepository)
+        {
+            _states = states.ToDictionary(s =>
+                StateNameReader.GetStateName(s.GetType()));
+            _contextRepository = contextRepository;
+        }
+
+        public void SetState(string state)
+        {
+            _state = _states[state];
+        }
+
+        public async Task DeleteContextAsync(string id)
+        {
```

Comment:
I am not sure that it is a good approach to delete the context in this way. How does mongodb execute the delete operation in case if there is no such entity? Does it throw an exception?